src/x86_64/service.c

```diff
@@ -237,27 +246,60 @@ static void __attribute__((noinline)) init_service_new_stack()
     heap misc = heap_general(kh);
     heap pages = heap_pages(kh);
 
+    /* runtime and console init */
+    init_debug("in init_service_new_stack");
+    unmap(0, PAGESIZE, pages);  /* unmap zero page */
     reclaim_regions();          /* unmap and reclaim stage2 stack */
+    init_debug("runtime");
     init_runtime(kh);
     init_extra_prints();
     init_pci(kh);
     init_console(kh);
+    init_symtab(kh);
+    read_kernel_syms();
+    init_debug("pci_discover (for VGA)");
     pci_discover(); // early PCI discover to configure VGA console
 
-    /* Unmap the first page so we catch faults on null pointer references. */
-    unmap(0, PAGESIZE, pages);
-
+    /* scheduling queues init */
     runqueue = allocate_queue(misc, 64);
     bhqueue = allocate_queue(misc, 2048); /* XXX will need something extensible really */
-    init_clock(kh);
+
+    /* interrupts */
+    init_debug("start_interrupts");
+    start_interrupts(kh);
+
+    /* platform detection and early init */
```

Comment:
Probably not something we need to worry about yet, but it's possible we'll run on instances with a combination of kvm and xen virtualization -- e.g., KVM as the core hypervisor doing CPU/EPT virtualization but also exposing xen PV devices. Just something to think about, in those cases we'll need to think about how to further refine our platform notion

Reply:
That is true. That is why I didn't create static notions of platforms and left it kind of loose. But since we don't have any such instances to support, it's not clear how even probing such a platform would work. Presently both KVM and Xen are detected via inspecting cpuid leaves - in fact each has a signature on the same leaf node, if I'm not mistaken. So I think we should punt on this for now.